src/Result.mo

```diff
@@ -85,53 +134,100 @@ public func fromOption<R, E>(x : ?R, err : E) : Result<R, E> {
   }
 };
 
-/// Maps the `Ok` type/value from the optional value, or else use the given error value.
-/// (Deprecate?)
-public func fromSomeMap<R1, R2, E>(x:?R1, f:R1->R2, err:E):Result<R2,E> {
-  switch x {
-    case (? x) {#ok (f x)};
-    case null {#err err};
+/// Create an option from a result, turning all #err into `null`.
+/// ```
+/// fromOption(#ok(x)) = ?x
+/// fromOption(#err(e)) = null
+/// ```
+public func toOption<R, E>(r : Result<R, E>) : ?R {
+  switch r {
+    case (#ok x) {?x};
+    case (#err _) {null};
   }
 };
 
-/// asserts that the option is Some(_) form.
-public func fromSome<Ok>(o:?Ok):Result<Ok,None> {
-  switch(o) {
-    case (?o) (#ok o);
-    case _ P.unreachable();
+/// Applies a function to a successful value, but discards the result. Use
+/// `iterate` if you're only interested in the side effect `f` produces.
+///
+/// ```
+/// var counter : Nat = 0;
+/// iterate<Nat, Text>(#ok(5), func (x : Nat) { counter += x });
+/// assert(counter == 5);
+/// iterate<Nat, Text>(#err("Wrong"), func (x : Nat) { counter += x });
+/// assert(counter == 5);
+/// ```
+public func iterate<Ok, Err>(res : Result<Ok, Err>, f : Ok -> ()) {
```

Comment:
Is this the name we use for other iters (In the Ocaml sense?) I don't recall.

Reply:
Yeah, we've decided on that name in `docs/design.md`